cppapi/server/multiattribute.cpp

```diff
@@ -786,26 +802,38 @@ void MultiAttribute::add_attribute(string &dev_name,DeviceClass *dev_class_ptr,l
 	{
 		if (idl_3 == false)
 		{
-			attr_list.push_back(new WAttribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new WAttribute(prop_list,attr,dev_name,index);
+			add_attr(new_attr);
 			index = attr_list.size() - 1;
 		}
 		else
 		{
-			attr_list.insert(ite,new WAttribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new WAttribute(prop_list,attr,dev_name,index);
+			attr_list.insert(ite,new_attr);
 			index = attr_list.size() - 3;
+			MultiAttributeExt::AttributePtrAndIndex mapElement;
+			mapElement.att_ptr = new_attr;
+			mapElement.att_index_in_vector = index;
+			ext->attr_map[new_attr->get_name_lower()] = mapElement;
 		}
 	}
 	else
 	{
 		if (idl_3 == false)
 		{
-			attr_list.push_back(new Attribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new Attribute(prop_list,attr,dev_name,index);
+			add_attr(new_attr);
 			index = attr_list.size() - 1;
 		}
 		else
 		{
-			attr_list.insert(ite,new Attribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new Attribute(prop_list,attr,dev_name,index);
```

Comment:
Seems to me as a code duplication. Must be extracted into a dedicated method overloaded from add_attr, see below